### PR TITLE
Set flat_fee per token network

### PR DESCRIPTION
**Pull request by palango**
Fixes: #4653 

## Description

This PR adds the possibility to set the flat mediation fee on a per token network basis. This is required as tokens in different networks have different values and therefore different settings are required.

This is accomplished by changing the `--flat-fee XXX` CLI flag to take two arguments: The checksummed address of the token network and the flat fee for that network (`--flat-fee 0xdeadbeef 123`). This flag can be set multiple times for different networks.

In the future we should introduce a `MediationFeeConfig` class to collect all related settings.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/utils/__init__.py [hunk 1]
```diff
@@ -68,9 +68,7 @@ def address_checksum_and_decode(addr: str) -> Address:
     if not is_checksum_address(addr):
         raise InvalidChecksummedAddress("Address must be EIP55 checksummed")
 
-    addr_bytes = decode_hex(addr)
-    assert len(addr_bytes) in (20, 0)
-    return Address(addr_bytes)
+    return to_canonical_address(addr)
```
Comment (palango) [inline raiden/utils/__init__.py]: This and below are just some drive by changes.

**Comment by LefterisJP** [comment]
> In the future we should introduce a MediationFeeConfig class to collect all related settings.

Definitely! This must be hard to set via the CLI in the long run.

File: raiden/ui/cli.py [hunk 2]
```diff
@@ -402,14 +401,16 @@ def options(func):
             "Mediation Fee Options",
             option(
                 "--flat-fee",
-                help=("Flat fee required for every mediation in wei of the mediated token."),
-                default=DEFAULT_MEDIATION_FLAT_FEE,
-                type=FeeAmount,
-                show_default=True,
+                help=(
```
Comment (LefterisJP) [inline raiden/ui/cli.py]: Do we have CLI tests that check that args are read correctly and passed on to the config we want?

If yes would be nice to have one for this argument, just to be sure it does what we think it does.

Comment (palango) [reply raiden/ui/cli.py]: Not that I'm aware of, but we should add these.

In this case it's quite nice that we can use `type=(ADDRESS_TYPE, click.IntRange(min=0)),`. This makes sure that there are exactly 2 arguments, and both of them get validated by the defined types.

**Pull request merged**